cirq/circuits/circuit.py

```diff
@@ -1570,22 +1574,26 @@ def to_text_diagram_drawer(
         diagram.write(0, 0, '')
         for q, i in qubit_map.items():
             diagram.write(0, i, qubit_namer(q))
-        if any(
-                isinstance(op, cirq.GlobalPhaseOperation)
-                for op in self.all_operations()):
-            diagram.write(0,
-                          max(qubit_map.values(), default=0) + 1,
-                          'global phase:')
+
+        # Print out global phase information
+        for op in self.all_operations():
+            if isinstance(op, cirq.GlobalPhaseOperation):
+                diagram.write(0,
+                              max(qubit_map.values(), default=0) + 1,
```

Comment:
Aside: Changing this to -1 would remove the extra '\n\n' for diagrams with only global phase.